compiler/rustc_middle/src/mir/interpret/error.rs

```diff
@@ -256,7 +256,10 @@ pub enum UndefinedBehaviorInfo<'tcx> {
     /// The value validity check found a problem.
     /// Should only be thrown by `validity.rs` and always point out which part of the value
     /// is the problem.
-    ValidationFailure(String),
+    ValidationFailure {
+        path: Option<String>,
```

Comment:
```suggestion
        /// The "path" to the value in question, e.g. `.0[5].field` for a struct
        /// field in the 5th element of an array that is the first element if a tuple.
        path: Option<String>,
```

Reply:
I've pushed a separate commit (to help with review) to add this documentation.